File: content/examples/other/leetcode/problems/meeting-rooms/example-1/solution.js
function canAttendMeetingsSort(intervals) {
  intervals.sort((a, b) => a[0] - b[0]);
  for (let i = 1; i < intervals.length; i += 1) {
    if (intervals[i][0] < intervals[i - 1][1]) return false;
  }
  return true;
}

if (require.main === module) {
  console.log(canAttendMeetingsSort([[0, 30], [5, 10], [15, 20]]));
  console.log(canAttendMeetingsSort([[7, 10], [2, 4]]));
}

module.exports = { canAttendMeetingsSort };

File: content/examples/other/leetcode/problems/meeting-rooms/example-1/Solution.java
import java.util.Arrays;

public class Solution {
  static boolean canAttendMeetingsSort(int[][] intervals) {
    Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
    for (int i = 1; i < intervals.length; i += 1) {
      if (intervals[i][0] < intervals[i - 1][1]) return false;
    }
    return true;
  }

  public static void main(String[] args) {
    System.out.println(canAttendMeetingsSort(new int[][] {{0, 30}, {5, 10}, {15, 20}}));
    System.out.println(canAttendMeetingsSort(new int[][] {{7, 10}, {2, 4}}));
  }
}
